### DISCUSSION TITLE
Not to accept unnecessary HL7 into Mirth

### DISCUSSION THREAD

**Discussion by imageRAD** [nextgenhealthcare/connect#6218] [Q&A] [answered]
Hi,

We have a channel listener for ORUs coming from vital monitors. Source is sending HL7 ZHB (kind of heart beat messages) frequently. So these messages are filling up the logs unnecessarily. We are giving NACKS and not to keep meta data however it increase message ID and stays there till next prune.

If there is a way to ignore these messages and not to have any trace in Mirth. We have found remove messages at client API however no remove message functionality inside Mirth.

Pre-processor also didn't work in this case or we couldn't find a way to make it happen.

Thx,

Haluk

**Comment by kirbykn2** [comment]
The pre-processor should work for this.  What did you try.
Here is some untested code.

// This script executes before the message is processed by the source connector
if (message.contains("ZHB|")) {
    // Discard the message by returning an empty string
    return "";
}

// Return the original message if it is not a ZHB message
return message;

**Comment by imageRAD** [comment]
Hi,

thx for the answer. "contains" didn't work however with "includes" did.  this is eliminating the content and messages are shown at the dashboard. we want not to have any messages on the dashboard and not to increase message ID. 

Cheers,

Haluk

<img width="847" alt="Screen Shot 2024-06-05 at 17 21 36" src="https://github.com/nextgenhealthcare/connect/assets/34233044/eb0bfce5-f770-4ba7-8a25-241e199b77da">
<img width="847" alt="Screen Shot 2024-06-05 at 17 21 28" src="https://github.com/nextgenhealthcare/connect/assets/34233044/6b4978cf-d7da-4c17-9fee-610e3fa73866">

**Reply by kirbykn2** [reply]
More untested code for you.  :)

```
// This script executes before the message is processed by the source connector
if (message.includes("ZHB|")) {
    // Return null to halt processing without logging or incrementing the message ID
    return null;
}

// Return the original message if it is not a ZHB message
return message;

```

**Reply by pacmano1** [reply]
Or just filter on another channel and send to the main channel

**Reply by imageRAD** [reply]
return null is also creating message ID and logged but clears the content of the messages. :(

**Reply by imageRAD** [reply]
> Or just filter on another channel and send to the main channel

yeap we thought this however number transactions too much this is not optimized way. and now not to create message ID becomes a challenge. :)

**Reply by kirbykn2** [reply]
What is the issue with increasing the msg id?

**Reply by pacmano1** [reply]
actually, filtering on another channel and storing no message history on that channel is certainly an optimized way.

**Reply by yonajj** [reply]
> Or just filter on another channel and send to the main channel

@imageRAD I think this the better option for what you want:

One channel with a high prunning, with a filter for those ZHB messages, and just a sender to another channel (Main)
Main channel will process only not ZHB message, and that message will have an Id that only will increment when a not ZHB message is received, and not high prunning is needed.

**Comment by kirbykn2** [comment]
The most optimized way is to have the sending system/device only send messages you need.   ;)

**Comment by pacmano1** [comment, marked as answer]
I think the final answer here is that @imageRAD 's request is not possible for a TCP/MLLP listener; however, another channel can filter messages, or the OP can stop worrying about extra messages that are filtered.
